pkg/kubernetes/utils.go

```diff
@@ -0,0 +1,44 @@
+package kubernetes
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	app "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func BuildObjectFromYamlFile(file string) (runtime.Object, error) {
+	yamlFile, err := ioutil.ReadFile(file)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	sch := runtime.NewScheme()
+	if err := scheme.AddToScheme(sch); err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	decoder := serializer.NewCodecFactory(sch).UniversalDeserializer().Decode
+
+	obj, _, err := decoder(yamlFile, nil, nil)
+	fmt.Println("<>", err)
+	return obj, err
+}
+
+func ConvertToConfigMap(obj runtime.Object) (*core.ConfigMap, error) {
+	return obj.(*core.ConfigMap), nil
+}
+
+func ConvertToDeployment(obj runtime.Object) (*app.Deployment, error) {
+	return obj.(*app.Deployment), nil
+}
```

Comment:
It’s better to do type check and return error.

Type assertion returns false if the interface is not a corresponding implementation.

Reply:
I updated!